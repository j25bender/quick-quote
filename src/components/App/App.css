html {
  background-color: #BFC0C0;
  height: 100%;
  padding-bottom: 40px; }
