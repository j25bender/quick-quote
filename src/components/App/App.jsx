import React, { Component } from 'react';
import './App.css';
import Header from '../Header/Header';
import Category from '../Category/Category';
import { Route, Switch } from 'react-router-dom';

export class App extends Component {
  render() {
    const routes = [ 'home', 'random', 'motivation', 'positive', 'life', 'funny', 'love', 'students', 'inspiration', 'meditation' ]; 
    const renderRoutes = routes.map( (route, i) => <Route key={ i } path={ '/' + route } component={ Category } /> );
    return (
      <div className="App" >
        <Header />
        <Switch>
          <Route exact path='/' component={ Category } />
          { renderRoutes }
        </Switch>
      </div>
    );
  }
}

export default App;