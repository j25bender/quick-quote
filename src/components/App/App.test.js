/* eslint-disable */
import React from 'react';
import App from './App';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App Component Test', () => {
  
  it('should render App component correctly', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});