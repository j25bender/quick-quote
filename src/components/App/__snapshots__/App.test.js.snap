// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Test should render App component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/home"
          />
          <Route
                    component={[Function]}
                    path="/random"
          />
          <Route
                    component={[Function]}
                    path="/motivation"
          />
          <Route
                    component={[Function]}
                    path="/positive"
          />
          <Route
                    component={[Function]}
                    path="/life"
          />
          <Route
                    component={[Function]}
                    path="/funny"
          />
          <Route
                    component={[Function]}
                    path="/love"
          />
          <Route
                    component={[Function]}
                    path="/students"
          />
          <Route
                    component={[Function]}
                    path="/inspiration"
          />
          <Route
                    component={[Function]}
                    path="/meditation"
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            Array [
              <Route
                component={[Function]}
                path="/home"
/>,
              <Route
                component={[Function]}
                path="/random"
/>,
              <Route
                component={[Function]}
                path="/motivation"
/>,
              <Route
                component={[Function]}
                path="/positive"
/>,
              <Route
                component={[Function]}
                path="/life"
/>,
              <Route
                component={[Function]}
                path="/funny"
/>,
              <Route
                component={[Function]}
                path="/love"
/>,
              <Route
                component={[Function]}
                path="/students"
/>,
              <Route
                component={[Function]}
                path="/inspiration"
/>,
              <Route
                component={[Function]}
                path="/meditation"
/>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/random",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/motivation",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/positive",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/life",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/funny",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/love",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/students",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/inspiration",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "9",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/meditation",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        path="/home"
            />
            <Route
                        component={[Function]}
                        path="/random"
            />
            <Route
                        component={[Function]}
                        path="/motivation"
            />
            <Route
                        component={[Function]}
                        path="/positive"
            />
            <Route
                        component={[Function]}
                        path="/life"
            />
            <Route
                        component={[Function]}
                        path="/funny"
            />
            <Route
                        component={[Function]}
                        path="/love"
            />
            <Route
                        component={[Function]}
                        path="/students"
            />
            <Route
                        component={[Function]}
                        path="/inspiration"
            />
            <Route
                        component={[Function]}
                        path="/meditation"
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              Array [
                <Route
                  component={[Function]}
                  path="/home"
/>,
                <Route
                  component={[Function]}
                  path="/random"
/>,
                <Route
                  component={[Function]}
                  path="/motivation"
/>,
                <Route
                  component={[Function]}
                  path="/positive"
/>,
                <Route
                  component={[Function]}
                  path="/life"
/>,
                <Route
                  component={[Function]}
                  path="/funny"
/>,
                <Route
                  component={[Function]}
                  path="/love"
/>,
                <Route
                  component={[Function]}
                  path="/students"
/>,
                <Route
                  component={[Function]}
                  path="/inspiration"
/>,
                <Route
                  component={[Function]}
                  path="/meditation"
/>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/random",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/motivation",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/positive",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/life",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/funny",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/love",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "7",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/students",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "8",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/inspiration",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "9",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/meditation",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
